src/FluentCheck.ts

```diff
@@ -175,58 +175,31 @@ class FluentCheckUniversal<K extends string, A, Rec extends ParentRec & Record<K
     for (const tp of collection) {
       testCase[this.name] = tp
       const result = callback(testCase)
-      if (!result.satisfiable) {
+      if (this.breakCondition(result.satisfiable)) {
         result.addExample(this.name, tp)
         return this.run(testCase, callback, result, depth + 1)
       }
     }
 
     return example
   }
-}
-
-class FluentCheckExistential<K extends string, A, Rec extends ParentRec & Record<K, A>, ParentRec extends {}>
-  extends FluentCheck<Rec, ParentRec> {
 
-  private cache: Array<FluentPick<A>>
-  private dedup: Arbitrary<A>
-
-  constructor(
-    protected readonly parent: FluentCheck<ParentRec, any>,
-    public readonly name: K,
-    public readonly a: Arbitrary<A>,
-    config: FluentConfig) {
-
-    super(parent, config)
-    this.dedup = a.unique()
-    this.cache = this.dedup.sampleWithBias(this.configuration.sampleSize)
-  }
-
-  protected run(
-    testCase: FluentPicks,
-    callback: (arg: FluentPicks) => FluentResult,
-    partial: FluentResult | undefined = undefined,
-    depth = 0): FluentResult {
+  abstract defaultValue: boolean
+  abstract breakCondition: (b: boolean) => boolean
```

Comment:
It looks simpler if we just have a `breakValue: boolean` here and do `result.satisfiable === breakValue`.